pkg/capabilities/consensus/ocr3/aggregators/reduce_aggregator.go

```diff
@@ -0,0 +1,436 @@
+package aggregators
+
+import (
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/shopspring/decimal"
+	"google.golang.org/protobuf/proto"
+
+	ocrcommon "github.com/smartcontractkit/libocr/commontypes"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/consensus/ocr3/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	"github.com/smartcontractkit/chainlink-common/pkg/values/pb"
+)
+
+type ReduceAggConfig struct {
+	// Configuration on how to aggregate one or more data points
+	Fields []AggregationField `mapstructure:"fields"  required:"true"`
+	// The top level field name that report data is put into
+	OutputFieldName string `mapstructure:"outputFieldName" json:"outputFieldName" default:"Reports"`
+	// The structure surrounding the report data that is put on to "OutputFieldName"
+	ReportFormat string `mapstructure:"reportFormat" json:"reportFormat" default:"map" jsonschema:"enum=map,enum=array"`
+}
+
+type AggregationField struct {
+	// The key to find a data point within the input data
+	// If omitted, the entire input will be used
+	InputKey string `mapstructure:"inputKey" json:"inputKey"`
+	// The key that the aggregated data is put under
+	// If omitted, the InputKey will be used
+	OutputKey string `mapstructure:"outputKey" json:"outputKey"`
+	// How the data set should be aggregated
+	Method string `mapstructure:"method" json:"method" jsonschema:"enum=median,enum=mode" required:"true"`
+	// An optional check to only report when the difference from the previous report exceeds a certain threshold.
+	// Can only be used when the field is of a numeric type: string, decimal, int64, big.Int, time.Time, float64
+	// If no deviation is provided on any field, there will always be a report once minimum observations are reached.
+	DeviationString string          `mapstructure:"deviation"  json:"deviation,omitempty"`
+	Deviation       decimal.Decimal `mapstructure:"-"  json:",omitempty"`
```

Comment:
Are we missing heartbeat here?

Reply:
This can be used as heartbeat by using an absolute deviation on the timestamp. I'll get some tests up to show that off.

Reply:
Yes exactly right on the config for PoR.

Agreed on needing more documentation. I'll create a design doc for this.